from PyQt6 import QtWidgets
from PyQt6 import QtCore
from PyQt6 import QtGui

from src.conDB import metody as metodyDB
from src.win.autoservisUsII import Ui_PomocAutoservis
from src.usApp import autoservis
import datetime


class Ui_AutoservisUs(Ui_PomocAutoservis, object):
    def setupUi(self, autoservisUs, email, autoser):
        """
        metoda na vytvoreni uvodniho okna autoservisu
        :param autoservisUs: autoservisu
        :param email: email prihlaseneho uzivatele
        :param autoser: list autoservisu
        :return: none
        """
        autoservisUs.setObjectName("autoservisUs")
        autoservisUs.resize(980, 680)
        self.autoservisUS = autoservisUs
        self.auta = metodyDB.nacteniAut()
        self.email = email
        self.cas = datetime.datetime.now()
        if len(autoser) == 0:
            self.autoservis = autoservis.Autoservisy.vypisAutoservis()
        else:
            self.autoservis = autoser

        self.centralwidget = QtWidgets.QWidget(parent=autoservisUs)
        self.centralwidget.setObjectName("centralwidget")
        self.gridLayout = QtWidgets.QGridLayout(self.centralwidget)
        self.gridLayout.setObjectName("gridLayout")
        self.tabWidget = QtWidgets.QTabWidget(parent=self.centralwidget)
        self.tabWidget.setObjectName("tabWidget")
        self.tab = QtWidgets.QWidget()
        self.tab.setObjectName("tab")
        self.gridLayout_3 = QtWidgets.QGridLayout(self.tab)
        self.gridLayout_3.setObjectName("gridLayout_3")
        spacerItem = QtWidgets.QSpacerItem(20, 40, QtWidgets.QSizePolicy.Policy.Minimum, QtWidgets.QSizePolicy.Policy.Expanding)
        self.gridLayout_3.addItem(spacerItem, 2, 1, 1, 1)
        self.label_3 = QtWidgets.QLabel(parent=self.tab)
        self.label_3.setObjectName("label_3")
        self.gridLayout_3.addWidget(self.label_3, 5, 1, 1, 2)
        self.comboAutoservis = QtWidgets.QComboBox(parent=self.tab)
        self.comboAutoservis.setObjectName("comboAutoservis")
        metodyDB.metoda1(self.autoservis, self.comboAutoservis)
        self.gridLayout_3.addWidget(self.comboAutoservis, 3, 3, 1, 1)
        self.horizontalLayout = QtWidgets.QHBoxLayout()
        self.horizontalLayout.setObjectName("horizontalLayout")
        self.rbtnServis = QtWidgets.QRadioButton(parent=self.tab)
        self.rbtnServis.setObjectName("rbtnServis")
        self.horizontalLayout.addWidget(self.rbtnServis)
        self.rbtnBouracka = QtWidgets.QRadioButton(parent=self.tab)
        self.rbtnBouracka.setLayoutDirection(QtCore.Qt.LayoutDirection.LeftToRight)
        self.rbtnBouracka.setAutoRepeatInterval(18)
        self.rbtnBouracka.setObjectName("rbtnBouracka")
        self.horizontalLayout.addWidget(self.rbtnBouracka)
        self.rbtnOprava = QtWidgets.QRadioButton(parent=self.tab)
        self.rbtnOprava.setObjectName("rbtnOprava")
        self.horizontalLayout.addWidget(self.rbtnOprava)
        self.gridLayout_3.addLayout(self.horizontalLayout, 7, 3, 1, 1)
        self.horizontalLayout_7 = QtWidgets.QHBoxLayout()
        self.horizontalLayout_7.setObjectName("horizontalLayout_7")
        self.label_11 = QtWidgets.QLabel(parent=self.tab)
        self.label_11.setAlignment(QtCore.Qt.AlignmentFlag.AlignLeading|QtCore.Qt.AlignmentFlag.AlignLeft|QtCore.Qt.AlignmentFlag.AlignVCenter)
        self.label_11.setObjectName("label_11")
        self.horizontalLayout_7.addWidget(self.label_11)
        self.rokVyroby = QtWidgets.QSpinBox(parent=self.tab)
        self.rokVyroby.setMinimum(2000)
        self.rokVyroby.setMaximum(self.cas.year)
        self.rokVyroby.setObjectName("rokVyroby")
        self.horizontalLayout_7.addWidget(self.rokVyroby)
        spacerItem1 = QtWidgets.QSpacerItem(40, 20, QtWidgets.QSizePolicy.Policy.Preferred, QtWidgets.QSizePolicy.Policy.Minimum)
        self.horizontalLayout_7.addItem(spacerItem1)
        self.label_12 = QtWidgets.QLabel(parent=self.tab)
        self.label_12.setAlignment(QtCore.Qt.AlignmentFlag.AlignRight|QtCore.Qt.AlignmentFlag.AlignTrailing|QtCore.Qt.AlignmentFlag.AlignVCenter)
        self.label_12.setObjectName("label_12")
        self.horizontalLayout_7.addWidget(self.label_12)
        self.najeteKm = QtWidgets.QSpinBox(parent=self.tab)
        self.najeteKm.setMaximum(999999)
        self.najeteKm.setObjectName("najeteKm")
        self.horizontalLayout_7.addWidget(self.najeteKm)
        self.gridLayout_3.addLayout(self.horizontalLayout_7, 6, 1, 1, 3)
        self.label = QtWidgets.QLabel(parent=self.tab)
        font = QtGui.QFont()
        font.setPointSize(20)
        self.label.setFont(font)
        self.label.setLayoutDirection(QtCore.Qt.LayoutDirection.LeftToRight)
        self.label.setAlignment(QtCore.Qt.AlignmentFlag.AlignCenter)
        self.label.setObjectName("label")
        self.gridLayout_3.addWidget(self.label, 1, 1, 1, 3)
        self.label_10 = QtWidgets.QLabel(parent=self.tab)
        self.label_10.setObjectName("label_10")
        self.gridLayout_3.addWidget(self.label_10, 7, 1, 1, 1)
        self.poznamka = QtWidgets.QTextEdit(parent=self.tab)
        self.poznamka.setMaximumSize(QtCore.QSize(16777215, 100))
        self.poznamka.setObjectName("poznamka")
        self.gridLayout_3.addWidget(self.poznamka, 9, 3, 1, 1)
        self.label_2 = QtWidgets.QLabel(parent=self.tab)
        self.label_2.setObjectName("label_2")
        self.gridLayout_3.addWidget(self.label_2, 8, 1, 1, 2)
        self.horizontalLayout_6 = QtWidgets.QHBoxLayout()
        self.horizontalLayout_6.setObjectName("horizontalLayout_6")
        spacerItem2 = QtWidgets.QSpacerItem(40, 20, QtWidgets.QSizePolicy.Policy.Expanding, QtWidgets.QSizePolicy.Policy.Minimum)
        self.horizontalLayout_6.addItem(spacerItem2)
        self.btnOdeslat = QtWidgets.QPushButton(parent=self.tab)
        self.btnOdeslat.setObjectName("btnOdeslat")
        self.horizontalLayout_6.addWidget(self.btnOdeslat)
        spacerItem3 = QtWidgets.QSpacerItem(40, 20, QtWidgets.QSizePolicy.Policy.Expanding, QtWidgets.QSizePolicy.Policy.Minimum)
        self.horizontalLayout_6.addItem(spacerItem3)
        self.gridLayout_3.addLayout(self.horizontalLayout_6, 11, 1, 1, 3)
        spacerItem4 = QtWidgets.QSpacerItem(20, 40, QtWidgets.QSizePolicy.Policy.Minimum, QtWidgets.QSizePolicy.Policy.Expanding)
        self.gridLayout_3.addItem(spacerItem4, 0, 1, 1, 3)
        self.horizontalLayout_3 = QtWidgets.QHBoxLayout()
        self.horizontalLayout_3.setObjectName("horizontalLayout_3")
        self.label_5 = QtWidgets.QLabel(parent=self.tab)
        self.label_5.setAlignment(QtCore.Qt.AlignmentFlag.AlignRight|QtCore.Qt.AlignmentFlag.AlignTrailing|QtCore.Qt.AlignmentFlag.AlignVCenter)
        self.label_5.setObjectName("label_5")
        self.horizontalLayout_3.addWidget(self.label_5)
        self.den = QtWidgets.QSpinBox(parent=self.tab)
        self.den.setMinimum(1)
        self.den.setMaximum(31)
        self.den.setObjectName("den")
        self.horizontalLayout_3.addWidget(self.den)
        self.den.setValue(self.cas.day)
        self.label_6 = QtWidgets.QLabel(parent=self.tab)
        self.label_6.setAlignment(QtCore.Qt.AlignmentFlag.AlignRight|QtCore.Qt.AlignmentFlag.AlignTrailing|QtCore.Qt.AlignmentFlag.AlignVCenter)
        self.label_6.setObjectName("label_6")
        self.horizontalLayout_3.addWidget(self.label_6)
        self.mesic = QtWidgets.QSpinBox(parent=self.tab)
        self.mesic.setMinimum(1)
        self.mesic.setMaximum(12)
        self.mesic.setObjectName("mesic")
        self.horizontalLayout_3.addWidget(self.mesic)
        self.mesic.setValue(self.cas.month)
        self.label_7 = QtWidgets.QLabel(parent=self.tab)
        self.label_7.setAlignment(QtCore.Qt.AlignmentFlag.AlignRight|QtCore.Qt.AlignmentFlag.AlignTrailing|QtCore.Qt.AlignmentFlag.AlignVCenter)
        self.label_7.setObjectName("label_7")
        self.horizontalLayout_3.addWidget(self.label_7)
        self.rok = QtWidgets.QSpinBox(parent=self.tab)
        self.rok.setMinimum(self.cas.year)
        self.rok.setMaximum(self.cas.year+50)
        self.rok.setObjectName("rok")
        self.horizontalLayout_3.addWidget(self.rok)
        self.gridLayout_3.addLayout(self.horizontalLayout_3, 8, 3, 1, 1)
        self.comboModel = QtWidgets.QComboBox(parent=self.tab)
        self.comboModel.setObjectName("comboModel")
        metodyDB.metoda1(self.auta, self.comboModel)
        self.gridLayout_3.addWidget(self.comboModel, 5, 3, 1, 1)
        self.label_9 = QtWidgets.QLabel(parent=self.tab)
        self.label_9.setObjectName("label_9")
        self.gridLayout_3.addWidget(self.label_9, 3, 1, 1, 1)
        self.label_4 = QtWidgets.QLabel(parent=self.tab)
        self.label_4.setObjectName("label_4")
        self.gridLayout_3.addWidget(self.label_4, 9, 1, 1, 2)
        spacerItem5 = QtWidgets.QSpacerItem(20, 40, QtWidgets.QSizePolicy.Policy.Minimum, QtWidgets.QSizePolicy.Policy.Expanding)
        self.gridLayout_3.addItem(spacerItem5, 12, 1, 1, 3)
        self.horizontalLayout_5 = QtWidgets.QHBoxLayout()
        self.horizontalLayout_5.setObjectName("horizontalLayout_5")
        spacerItem6 = QtWidgets.QSpacerItem(40, 20, QtWidgets.QSizePolicy.Policy.Expanding, QtWidgets.QSizePolicy.Policy.Minimum)
        self.horizontalLayout_5.addItem(spacerItem6)
        self.chybHlaska = QtWidgets.QLabel(parent=self.tab)
        self.chybHlaska.setText("")
        self.chybHlaska.setObjectName("chybHlaska")
        self.horizontalLayout_5.addWidget(self.chybHlaska)
        spacerItem7 = QtWidgets.QSpacerItem(40, 20, QtWidgets.QSizePolicy.Policy.Expanding, QtWidgets.QSizePolicy.Policy.Minimum)
        self.horizontalLayout_5.addItem(spacerItem7)
        self.gridLayout_3.addLayout(self.horizontalLayout_5, 10, 1, 1, 3)
        spacerItem8 = QtWidgets.QSpacerItem(40, 20, QtWidgets.QSizePolicy.Policy.Preferred, QtWidgets.QSizePolicy.Policy.Minimum)
        self.gridLayout_3.addItem(spacerItem8, 8, 0, 1, 1)
        spacerItem9 = QtWidgets.QSpacerItem(40, 20, QtWidgets.QSizePolicy.Policy.Preferred, QtWidgets.QSizePolicy.Policy.Minimum)
        self.gridLayout_3.addItem(spacerItem9, 7, 4, 1, 1)
        self.tabWidget.addTab(self.tab, "")
        self.tab_2 = QtWidgets.QWidget()
        self.tab_2.setObjectName("tab_2")
        self.gridLayout_2 = QtWidgets.QGridLayout(self.tab_2)
        self.gridLayout_2.setObjectName("gridLayout_2")
        spacerItem10 = QtWidgets.QSpacerItem(40, 20, QtWidgets.QSizePolicy.Policy.Fixed, QtWidgets.QSizePolicy.Policy.Minimum)
        self.gridLayout_2.addItem(spacerItem10, 1, 3, 1, 1)
        self.horizontalLayout_8 = QtWidgets.QHBoxLayout()
        self.horizontalLayout_8.setObjectName("horizontalLayout_8")
        spacerItem11 = QtWidgets.QSpacerItem(40, 20, QtWidgets.QSizePolicy.Policy.Expanding, QtWidgets.QSizePolicy.Policy.Minimum)
        self.horizontalLayout_8.addItem(spacerItem11)
        self.label_13 = QtWidgets.QLabel(parent=self.tab_2)
        self.label_13.setObjectName("label_13")
        self.horizontalLayout_8.addWidget(self.label_13)
        self.idVyhledat = QtWidgets.QSpinBox(parent=self.tab_2)
        self.idVyhledat.setMaximum(99999)
        self.idVyhledat.setObjectName("idVyhledat")
        self.horizontalLayout_8.addWidget(self.idVyhledat)
        self.btnVyhledat = QtWidgets.QPushButton(parent=self.tab_2)
        self.btnVyhledat.setObjectName("btnVyhledat")
        self.horizontalLayout_8.addWidget(self.btnVyhledat)
        spacerItem12 = QtWidgets.QSpacerItem(40, 20, QtWidgets.QSizePolicy.Policy.Expanding, QtWidgets.QSizePolicy.Policy.Minimum)
        self.horizontalLayout_8.addItem(spacerItem12)
        self.gridLayout_2.addLayout(self.horizontalLayout_8, 5, 1, 1, 2)
        spacerItem13 = QtWidgets.QSpacerItem(20, 10, QtWidgets.QSizePolicy.Policy.Minimum, QtWidgets.QSizePolicy.Policy.Fixed)
        self.gridLayout_2.addItem(spacerItem13, 0, 1, 1, 2)
        spacerItem14 = QtWidgets.QSpacerItem(40, 20, QtWidgets.QSizePolicy.Policy.Fixed, QtWidgets.QSizePolicy.Policy.Minimum)
        self.gridLayout_2.addItem(spacerItem14, 1, 0, 1, 1)
        self.tabulka = QtWidgets.QTableView(parent=self.tab_2)
        self.tabulka.setObjectName("tabulka")
        self.gridLayout_2.addWidget(self.tabulka, 3, 1, 1, 2)
        spacerItem15 = QtWidgets.QSpacerItem(20, 10, QtWidgets.QSizePolicy.Policy.Minimum, QtWidgets.QSizePolicy.Policy.Fixed)
        self.gridLayout_2.addItem(spacerItem15, 2, 1, 1, 2)
        self.chybHlaska2 = QtWidgets.QLabel(parent=self.tab_2)
        self.chybHlaska2.setAlignment(QtCore.Qt.AlignmentFlag.AlignCenter)
        self.chybHlaska2.setObjectName("chybHlaska2")
        self.gridLayout_2.addWidget(self.chybHlaska2, 4, 2, 1, 1)
        self.horizontalLayout_9 = QtWidgets.QHBoxLayout()
        self.horizontalLayout_9.setObjectName("horizontalLayout_9")
        spacerItem16 = QtWidgets.QSpacerItem(40, 20, QtWidgets.QSizePolicy.Policy.Maximum, QtWidgets.QSizePolicy.Policy.Minimum)
        self.horizontalLayout_9.addItem(spacerItem16)
        self.label_8 = QtWidgets.QLabel(parent=self.tab_2)
        self.label_8.setMinimumSize(QtCore.QSize(0, 0))
        font = QtGui.QFont()
        font.setPointSize(22)
        self.label_8.setFont(font)
        self.label_8.setAlignment(QtCore.Qt.AlignmentFlag.AlignCenter)
        self.label_8.setObjectName("label_8")
        self.horizontalLayout_9.addWidget(self.label_8)
        spacerItem17 = QtWidgets.QSpacerItem(40, 20, QtWidgets.QSizePolicy.Policy.Expanding, QtWidgets.QSizePolicy.Policy.Minimum)
        self.horizontalLayout_9.addItem(spacerItem17)
        self.btnRefresh = QtWidgets.QPushButton(parent=self.tab_2)
        self.btnRefresh.setFlat(True)
        self.btnRefresh.setObjectName("btnRefresh")
        self.horizontalLayout_9.addWidget(self.btnRefresh)
        self.gridLayout_2.addLayout(self.horizontalLayout_9, 1, 2, 1, 1)
        self.tabWidget.addTab(self.tab_2, "")
        self.gridLayout.addWidget(self.tabWidget, 1, 0, 1, 5)
        self.btnOdhlasitse = QtWidgets.QPushButton(parent=self.centralwidget)
        self.btnOdhlasitse.setObjectName("btnOdhlasitse")
        self.gridLayout.addWidget(self.btnOdhlasitse, 0, 4, 1, 1)
        self.btnAutoservisy = QtWidgets.QPushButton(parent=self.centralwidget)
        self.btnAutoservisy.setObjectName("btnAutoservisy")
        self.gridLayout.addWidget(self.btnAutoservisy, 0, 3, 1, 1)
        self.btnUs = QtWidgets.QPushButton(parent=self.centralwidget)
        self.btnUs.setAutoDefault(False)
        self.btnUs.setDefault(False)
        self.btnUs.setFlat(True)
        self.btnUs.setObjectName("btnUs")
        self.gridLayout.addWidget(self.btnUs, 0, 2, 1, 1)
        spacerItem18 = QtWidgets.QSpacerItem(40, 20, QtWidgets.QSizePolicy.Policy.Expanding, QtWidgets.QSizePolicy.Policy.Minimum)
        self.gridLayout.addItem(spacerItem18, 0, 0, 1, 1)
        self.usjmeno = QtWidgets.QLabel(parent=self.centralwidget)
        self.usjmeno.setObjectName("usjmeno")
        self.gridLayout.addWidget(self.usjmeno, 0, 1, 1, 1)
        autoservisUs.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(parent=autoservisUs)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 960, 24))
        self.menubar.setObjectName("menubar")
        autoservisUs.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(parent=autoservisUs)
        self.statusbar.setObjectName("statusbar")
        autoservisUs.setStatusBar(self.statusbar)
        self.btnOdhlasitse.clicked.connect(self.ui_prihlaseni)
        self.btnUs.clicked.connect(lambda : self.ui_uzivatel(email))
        self.btnAutoservisy.clicked.connect(self.zobrazAutoser)
        self.btnOdeslat.clicked.connect(self.akceOdeslani)
        self.btnVyhledat.clicked.connect(self.hledatAkce)
        self.btnRefresh.clicked.connect(self.obnovaAkce)
        self.obnovaAkce()
        self.retranslateUi(autoservisUs, email)
        self.tabWidget.setCurrentIndex(0)

        QtCore.QMetaObject.connectSlotsByName(autoservisUs)


def zobrazHlavniMenu(object : object, email, autoservisy):
    """
    metoda na zobrazeni okna autoservisu
    :param object: objekt
    :param email: email uzivatele
    :param autoservisy: list autoservisu
    :return: none
    """
    object.autoservisUs = QtWidgets.QMainWindow()
    object.ui = Ui_AutoservisUs()
    object.ui.setupUi(object.autoservisUs, email, autoservisy)
    object.autoservisUs.show()